File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    var ans=0,mn=1000000;
    for(x in prices)
    {
         mn=Math.min(mn,prices[x]);
         ans=Math.max(ans,prices[x]-mn);
    }
    return ans;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int mn=prices[0],ans=0;
        for(int x:prices)
        {
            mn=Math.min(mn,x);
            ans=Math.max(ans,x-mn);
        }
        return ans;
    }
}